lib/remap-functions/src/includes.rs

```diff
@@ -50,11 +50,22 @@ impl Expression for IncludesFn {
     fn type_def(&self, state: &state::Compiler) -> TypeDef {
         use value::Kind;
 
-        self.value
-            .type_def(state)
-            .fallible_unless(Kind::Array)
-            .merge(self.item.type_def(state))
-            .with_constraint(Kind::Boolean)
+        let value_type_def = self.value.type_def(state).fallible_unless(Kind::Array);
+
+        let value_fallible = value_type_def.is_fallible();
+
+        let inner_kind = value_type_def
+            .inner_type_def
+            .unwrap_or_default()
+            .scalar_kind();
+
+        let item_type_def = self.item.type_def(state).fallible_unless(inner_kind);
```

Comment:
This bit isn't right. Although it might be pointles to call `includes("foo", [1 ,2, 3])` it won't cause any error, so I don't think we need to check against the type.

Probably:

```suggestion
        let item_type_def = self.item.type_def(state);
```

Reply:
Aha, gotcha, I didn't realize arrays could be multiple types, but that makes sense. I'll remove that check.